consensus/core/src/tx.rs

```diff
@@ -30,27 +35,57 @@ pub use smallvec::smallvec as scriptvec;
 pub type ScriptPublicKeys = HashSet<ScriptPublicKey>;
 
 /// Represents a Kaspad ScriptPublicKey
-#[derive(Default, Debug, PartialEq, Eq, Serialize, Clone, Hash)]
-#[serde(rename_all = "camelCase")]
+#[derive(Default, Debug, PartialEq, Eq, Clone, Hash)]
 pub struct ScriptPublicKey {
     version: ScriptPublicKeyVersion,
     script: ScriptVec, // Kept private to preserve read-only semantics
 }
 
+#[derive(Default, Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Hash)]
+#[serde(rename_all = "camelCase")]
+struct ScriptPublicKeyInternal<'a> {
+    version: ScriptPublicKeyVersion,
+    script: &'a [u8],
+}
+
+impl Serialize for ScriptPublicKey {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        if serializer.is_human_readable() {
+            let mut hex = [0u8; SCRIPT_VECTOR_SIZE * 2 + 4];
+            faster_hex::hex_encode(&self.version.to_be_bytes(), &mut hex).expect("Unable to serialize ScriptPubKey version");
+            faster_hex::hex_encode(&self.script, &mut hex[4..]).expect("Unable to serialize ScriptPubKey script");
+            serializer.serialize_str(str::from_utf8(&hex).expect("hex is must be a valid UTF-8"))
+        } else {
+            ScriptPublicKeyInternal { version: self.version, script: &self.script }.serialize(serializer)
```

Comment:
```
let mut state = serializer.serialize_struct("ScriptPublicKey", 2)?;
            state.serialize_field("version", &self.version)?;
            state.serialize_field("script", self.script.as_ref())?;
            state.end()
 ```
 it can be done without `ScriptPublicKeyInternal`, but I dont think it's a big deal

Reply:
let's keep it as-is